src/utils/daos_metrics/daos_metrics.c

```diff
@@ -0,0 +1,211 @@
+	       "\tDefault is 1 second\n"
+	       "--help, -h\n"
+	       "\tThis help text\n\n"
+	       "Customize the displayed data by specifying one or more "
+	       "filters:\n"
+	       "\tDefault is include everything\n\n"
+	       "--counter, -c\n"
+	       "\tInclude counters\n"
+	       "--duration, -d\n"
+	       "\tInclude durations\n"
+	       "--timestamp, -t\n"
+	       "\tInclude timestamps\n"
+	       "--snapshot, -s\n"
+	       "\tInclude timer snapshots\n"
+	       "--gauge, -g\n"
+	       "\tInclude gauges\n",
+	       prog_name);
+}
+
+int
+main(int argc, char **argv)
+{
+	struct d_tm_node_t	*root = NULL;
+	struct d_tm_node_t	*node = NULL;
+	uint64_t		*shmem_root = NULL;
+	char			dirname[D_TM_MAX_NAME_LEN] = {0};
+	int			rank = 0;
+	int			iteration = 0;
+	int			num_iter = 0;
+	int			filter = 0;
+	int			delay = 1;
+	int			c;
+
+	sprintf(dirname, "/");
+
+	/********************* Parse user arguments *********************/
+	while (1) {
+		static struct option long_options[] = {
+			{"rank", required_argument, NULL, 'r'},
+			{"counter", no_argument, NULL, 'c'},
+			{"duration", no_argument, NULL, 'd'},
+			{"timestamp", no_argument, NULL, 't'},
+			{"snapshot", no_argument, NULL, 's'},
+			{"gauge", no_argument, NULL, 'g'},
+			{"iterations", required_argument, NULL, 'i'},
+			{"path", required_argument, NULL, 'p'},
+			{"delay", required_argument, NULL, 'D'},
+			{"help", no_argument, NULL, 'h'},
+			{NULL, 0, NULL, 0}
+		};
+
+		c = getopt_long_only(argc, argv, "r:cdtsgi:p:D:h",
+				     long_options, NULL);
+		if (c == -1)
+			break;
+
+		switch (c) {
+		case 'r':
+			rank = atoi(optarg);
+			break;
+		case 'c':
+			filter |= D_TM_COUNTER;
+			break;
+		case 'd':
+			filter |= D_TM_DURATION;
+			break;
+		case 't':
+			filter |= D_TM_TIMESTAMP;
+			break;
+		case 's':
+			filter |= D_TM_TIMER_SNAPSHOT;
+			break;
+		case 'g':
+			filter |= D_TM_GAUGE;
+			break;
+		case 'i':
+			num_iter = atoi(optarg);
+			break;
+		case 'p':
+			snprintf(dirname, sizeof(dirname), "%s", optarg);
+			break;
+		case 'D':
+			delay = atoi(optarg);
+			break;
+		case 'h':
+		case '?':
+		default:
+			print_usage(argv[0]);
+			exit(0);
+		}
+	}
+
+	if (filter == 0)
+		filter = D_TM_COUNTER | D_TM_DURATION | D_TM_TIMESTAMP |
+			 D_TM_TIMER_SNAPSHOT | D_TM_GAUGE;
+
+	shmem_root = d_tm_get_shared_memory(rank);
+	if (!shmem_root)
+		goto failure;
+
+	root = d_tm_get_root(shmem_root);
+	if (!root)
+		goto failure;
+
+	if (strncmp(dirname, "/", D_TM_MAX_NAME_LEN) != 0) {
+		node = d_tm_find_metric(shmem_root, dirname);
+		if (node != NULL) {
+			root = node;
+		} else {
+			printf("No metrics found at: '%s'\n",
+			       dirname);
+			exit(0);
+		}
+	}
+
+	while ((num_iter == 0) || (iteration < num_iter)) {
+		printf("\nIteration: %d\n", iteration);
+		printf("There are %" PRIu64 " matching metrics underneath the"
+		       " directory %s\n",
+		       d_tm_count_metrics(shmem_root, root, filter),
+		       dirname[0] != '\0' ? dirname : "/");
```

Comment:
I'm not sure this output adds much from a user perspective. The way I would use this is to run the tool using watch, e.g.

`$ watch -n1 daos_metrics -p server -i1`

In this case the iteration count is always 0, but even if a developer isn't using watch, the iteration and matching metrics count just seems to add visual noise to the output.

Reply:
That makes sense to me.  I'll remove those two statements.